openlibrary/plugins/upstream/account.py

```diff
@@ -403,6 +405,11 @@ def POST(self):
             config.login_cookie_name, web.ctx.conn.get_auth_token(), expires=expires
         )
         ol_account = OpenLibraryAccount.get(email=email)
+
+        # Don't overwrite the cookie, which will contain banner display preferences
+        if not web.cookies().get('se', False):
+            self.set_screener_cookie(ol_account)
+
```

Comment:
We are determining if a patron is eligible for the survey whenever they log in.  The problem with this approach is that people who chose the "Remember me" option while logging in will likely not see the banner.
Not sure how to solve this issue.  Do we have a means of logging everybody out at once?  Doing so before the campaign would ensure that everybody who is eligible and visits will see the banner, but would also be an annoyance for our patrons.
Maybe we set the screener cookie on the My Books page instead of at login?  This would avoid logging everybody out, but we may still miss eligible folks who seldomly visit their My Books page.

Reply:
This is a good point. If we do a deploy that might log people out? For now, maybe it's okay if we ignore this case (let's see how many patrons reply). It's a really good point and there are ways to log everyone out.